src/taskRunner.ts

```diff
@@ -0,0 +1,82 @@
+import { CustomExecution, Event, EventEmitter, Pseudoterminal, Task, tasks, TaskScope } from 'vscode';
+import { d2Tasks } from './tasks';
+
+// eslint-disable-next-line no-unused-vars
+export type TaskRunnerCallback = (data: string) => void
+// eslint-disable-next-line no-unused-vars
+export type TaskOutput = (text: string) => void
+
+/**
+ * TaskRunner - This creates CustomTask and Pseudotermial to run it in.
+ * This will return immediately, the tasks callback should be called 
+ * when the task is finished.
+ */
+export class TaskRunner {
+
+    public retVal = '';
+    public genTask(filename: string, text: string, callback: TaskRunnerCallback): void {
+        const pty = new CustomTaskTerminal(filename, text, callback);
+        const ce = new CustomExecution((): Promise<CustomTaskTerminal> =>
+            new Promise((resolve) => {
+                resolve(pty);
+            }));
+
+        const task = new Task(
+            { type: "D2 Type" },
```

Comment:
```suggestion
            { type: "D2" },
```

I think we need to register the task type https://code.visualstudio.com/api/extension-guides/task-provider#task-provider